src/WebSocketsServer.cpp

```diff
@@ -118,7 +126,29 @@ void WebSocketsServer::onEvent(WebSocketServerEvent cbEvent) {
     _cbEvent = cbEvent;
 }
 
-/**
+/*
+ * Sets the custom http header validator function
+ * @param httpHeaderValidationFunc WebSocketServerHttpHeaderValFunc ///< pointer to the custom http header validation function
+ * @param mandatoryHttpHeaders const char* ///< the array of named http headers considered to be mandatory / must be present in order for websocket upgrade to succeed
+ */
+void WebSocketsServer::onValidateHttpHeader(
+	WebSocketServerHttpHeaderValFunc validationFunc,
+	const char* mandatoryHttpHeaders[])
+{
+	_httpHeaderValidationFunc = validationFunc;
+
+	if (_mandatoryHttpHeaders)
+		delete[] _mandatoryHttpHeaders;
+
+	_mandatoryHttpHeaderCount = (sizeof(mandatoryHttpHeaders) / sizeof(char*));

```

Comment:
this is really working for a pointer to a array?
may I am wrong, but i think ```sizeof(mandatoryHttpHeaders)``` will always return 4 (the same as ```sizeof(char*)```.
since all pointers on ESP are 32bit.